File: src/main/java/com/example/demoimdb/model/Episode.java
package com.example.demoimdb.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;

import javax.persistence.*;
import java.util.List;

@Entity
@Table(name = "episode")
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class Episode {
    @ManyToOne
    @JoinColumn(name = "movie_id")
    @JsonIgnore
    private Movie movie;
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(name = "ep")
    private Integer ep;
    @Column(name = "season")
    private Integer season;
    @Column(name = "name")
    private String name;
    @Column(name = "description", length = 1000)
    private String description;
    @Column(name = "image")
    private String image;
    @Column(name = "releaseDate")
    private String releaseDate;
    @Column(name = "duration")
    private String duration;
    @Column(name = "number_vote")
    private Integer numberVote;
    @Column(name = "score")
    private Double score;
    @OneToMany(mappedBy = "episode")
    private List<Comment> listComments;
    @OneToMany(mappedBy = "episode")
    private List<Rating> listRatings;
}

File: src/entities/Episode.ts
import {
  Column,
  Entity,
  ManyToOne,
  OneToMany,
  PrimaryGeneratedColumn,
} from 'typeorm';
import { Comment } from './Comment';
import { Movie } from './Movie';
import { Rating } from './Rating';

@Entity({ name: 'episodes' })
export class Episode {
  @PrimaryGeneratedColumn()
  id: number;

  @Column()
  ep: number;

  @Column()
  season: number;

  @Column()
  name: string;

  @Column({ nullable: true, length: 1000 })
  description: string;

  @Column({ nullable: true })
  image: string;

  @Column({ nullable: true })
  releaseDate: string;

  @Column({ nullable: true })
  duration: number;

  @Column({ nullable: true })
  numberVote: number;

  @Column({ nullable: true, type: 'float' })
  score: number;

  @ManyToOne(() => Movie, (movie) => movie.episodes, { onDelete: 'SET NULL' })
  movie: Movie;

  @OneToMany(() => Rating, (rating) => rating.episode)
  ratings: Rating[];

  @OneToMany(() => Comment, (comment) => comment.episode)
  comments: Comment[];
}
